crates/hir-ty/src/tests/never_type.rs

```diff
@@ -539,3 +539,249 @@ fn test() {
 "#,
     );
 }
+
```

Comment:
These tests are all of the tests that included in the rustc's original PR, except one case (commented below) and mir ones